test/E2ETest/TyeRunTests.cs

```diff
@@ -1011,6 +1011,44 @@ public async Task MultiRepo_WorksWithCloning()
             });
         }
 
+        [ConditionalFact]
+        [SkipIfDockerNotRunning]
+        public async Task MultiRepo_WorksWithCloningAndDockerfile()
+        {
+            using var projectDirectory = TempDirectory.Create(preferUserDirectoryOnMacOS: true);
+
+            var content = @"
+name: tye-docker-sample
+services:
+- name: minapp
+  repository: https://github.com/OlegKarasik/tye-docker-sample";
```

Comment:
@davidfowl, I understand the issue with the repo. I personally have no plans to remove it but in any case, may be there is a way to fork it into more persistent (and maintainable by the team) place? I am good with this.

Reply:
I wonder if it's possible to do this within the tye repo itself?

Reply:
@davidfowl, I don't think this is possible without introducing changes to how `tye` searches for configuration files. Currently this is done in the `ConfigFileFinder` class, which does search in the root of the passed directory:

```csharp
public static bool TryFindSupportedFile(
  string directoryPath, 
  [NotNullWhen(true)] out string? filePath, 
  [MaybeNullWhen(true)] out string? errorMessage, 
  string[]? fileFormats = null)
{
    fileFormats ??= FileFormats;
    foreach (var format in fileFormats)
    {
        var files = Directory.GetFiles(directoryPath, format);

        switch (files.Length)
        {
            case 1:
                errorMessage = null;
                filePath = files[0];
                return true;
            case 0:
                continue;
        }

        errorMessage = $"More than one matching file was found in directory '{directoryPath}'.";
        filePath = default;
        return false;
    }

    errorMessage = $"No project project file or solution was found in directory '{directoryPath}'.";
    filePath = default;
    return false;
}
```

This method is used to find a configuration file in the directory of cloned repository inside `ApplicationFactory` class:

```csharp
var clonePath = Path.Combine(rootConfig.Source.DirectoryName!, path, configService.Name);

if (!Directory.Exists(clonePath))
{
  if (!await GitDetector.Instance.IsGitInstalled.Value)
  {
      throw new CommandException($"Cannot clone repository {configService.Repository} because git is not installed. Please install git if you'd like to use \"repository\" in tye.yaml.");
  }

  var result = await ProcessUtil.RunAsync("git", $"clone {configService.Repository} \"{clonePath}\"", workingDirectory: rootConfig.Source.DirectoryName, throwOnError: false);

  if (result.ExitCode != 0)
  {
      throw new CommandException($"Failed to clone repository {configService.Repository} with exit code {result.ExitCode}.{Environment.NewLine}{result.StandardError}{result.StandardOutput}.");
  }
}

if (!ConfigFileFinder.TryFindSupportedFile(clonePath, out var file, out var errorMessage))
{
  throw new CommandException(errorMessage!);
}
```